NorthstarDLL/plugins/plugin_abi.h

```diff
@@ -114,32 +110,7 @@ struct PluginEngineData
 	void* conVarRegister;
 	void* ConVar_Vtable;
 	void* IConVar_Vtable;
-};
-
-struct PluginGameStatePresence
-{
-	const char* id;
-	const char* name;
-	const char* description;
-	const char* password;
-
-	bool isServer;
-	bool isLocal;
-	GameState state;
-
-	const char* map;
-	const char* mapDisplayname;
-	const char* playlist;
-	const char* playlistDisplayname;
-
-	int currentPlayers;
-	int maxPlayers;
-
-	int ownScore;
-	int otherHighestScore; // NOTE: The highest score OR the second highest score if we have the highest
-	int maxScore;
-
-	int timestampEnd;
+	void* g_pCVar;
```

Comment:
All of the data functions should have their type information exposed, just like the init funcs does.

plugins that directly load types from NorthstarLauncher could then make reasonable decisions without having to literally copy a dozen files over.

Reply:
(this is not the code I selected, go home github you are drunk)

Reply:
I hate header files

Reply:
I have no idea how to make the compiler not complain about duplicate types

Reply:
You could rename the typedefs to something else.
Or maybe this doesn't need to be done, since Titanfall 2 and N* are built using MSVC and MSVC is not ABI stable this could cause more issues than its worth.

Reply:
resolved by 6195047